### PR TITLE
CB-8455 added --nohooks option

### PR DESCRIPTION
**Pull request by bso-intel**
implemented --nohooks noption as described in CB-8455

### CODE REVIEW

**Comment by purplecabbage** [comment]
This feature has much deeper and far reaching requirements, as the CI has pointed out.
The individual platforms are all keeping their own parsers, and because this affects the platform-api ALL platforms must be updated for this to work.

**Comment by bso-intel** [comment]
Hi Jesse,

The CI failures actually were caused by not checking the existence of the opts parameter.
In the spec unit tests, update_project() was invoked without the opts parameter that I added in windows_parser.js and wp8_parser.js.
I have fixed the issues, and now all tests passed.

**Comment by vladimir-kotikov** [comment]
Isn't hooks is a CLI/LIB feature? As far as I remember, hooks fired by LIB except for `pre_package` hook for wp8/windows, so probably there is no need to pass the option down to platforms.

**Comment by bso-intel** [comment]
Hi Vladmir,

Exactly.
The pre-package hooks on for the windows* platforms run when LIB prepares for the source.
To suppress this hook, HookRunner needs the option "--nohooks pre_package" that is propagated from the command line.

**Comment by bso-intel** [comment]
HI Jesse and Vladimir,
If there is no more comment, could you pull my branch CB-8455 so the next Cordova release contains this feature?
Thanks.

File: cordova-lib/src/hooks/HooksRunner.js [hunk 1]
```diff
@@ -87,6 +93,12 @@ module.exports = HooksRunner;
  */
 module.exports.fire = globalFire;
 function globalFire(hook, opts) {
+    if (isHookDisabled(opts, hook)) {
+        /* jshint -W064 */
```
Comment (vladimir-kotikov) [inline cordova-lib/src/hooks/HooksRunner.js]: Just a minor nit: do we need for this ignores? We don't have neither 'strict mode' nor JSHint `newcap` option enabled, so this is probably just a no-op.

**Comment by vladimir-kotikov** [comment]
Looks good.

**Comment by bso-intel** [comment]
Hi Vladimir,

Thank you very much for reviewing.
I addressed your feedback and added several unit tests in LIB and CLI.

**Comment by bso-intel** [comment]
Thank you very much for merging, Vladimir.

**Pull request closed without merging**